src/phoenix/server/api/mutations/auth.py

```diff
@@ -3,16 +3,49 @@
 from strawberry import Info
 from strawberry.permission import BasePermission
 
+from phoenix.server.bearer_auth import PhoenixUser
+
+
+class IsNotReadOnly(BasePermission):
+    message = "Application is read-only"
+
+    def has_permission(self, source: Any, info: Info, **kwargs: Any) -> bool:
+        return not info.context.read_only
+
 
 class IsAuthenticated(BasePermission):
     message = "User is not authenticated"
 
-    async def has_permission(self, source: Any, info: Info, **kwargs: Any) -> bool:
-        return not info.context.read_only
+    def has_permission(self, source: Any, info: Info, **kwargs: Any) -> bool:
+        if info.context.token_store is None:
+            return True
+        try:
+            user = info.context.request.user
+        except AttributeError:
+            return False
+        return isinstance(user, PhoenixUser) and user.is_authenticated
+
+
+class IsAdmin(BasePermission):
+    message = "Only admin can perform this action"
+
+    def has_permission(self, source: Any, info: Info, **kwargs: Any) -> bool:
+        if info.context.token_store is None:
+            return False
+        try:
+            user = info.context.request.user
+        except AttributeError:
+            return False
+        return (
+            isinstance(user, PhoenixUser)
+            and user.is_authenticated
+            and user.claim.attributes is not None
+            and user.claim.attributes.user_role == "ADMIN"
```

Comment:
Non blocking but we should probably try to avoid repeating roles as strings like "ADMIN" at some point as it feels a bit brittle.

Reply:
ok, I can also add an enum for now